### DISCUSSION TITLE
Need some help in setting up the project in my local system

### DISCUSSION THREAD

**Discussion by niteshmodi** [Atri-Labs/atrilabs-engine#198] [Help: Installation & Start] [answered]
Can someone help me in setting up the project in my local machine 
![question1](https://user-images.githubusercontent.com/89907820/190704679-36c07722-d9e9-4d70-aa60-696a562aa3a3.png)

(After reading the contribution.md file )
when I tried to use the export command in my terminal it is not working I successfully forked using the clone commands and installed yarn but I got stuck after the install yarn command process

**Comment by niteshmodi** [comment]
![question2](https://user-images.githubusercontent.com/89907820/190709100-8b9ffadc-89ff-4763-b0ab-23aa89394b6c.png)
i am having problem in all this steps I think the commands are not for windows I tried using "set' instead of "export" but even after that got some problems in next steps 

![q3](https://user-images.githubusercontent.com/89907820/190709290-d2814e84-60e8-44fe-99fc-c3c9b7382ac2.png)
errors like this 

please help

**Comment by cruxcode** [comment]
Hi @niteshmodi,

Yes, the build steps are not for Windows. We will add steps for windows in the CONTRIBUTING.md soon.

Sorry for the inconveniences.

**Comment by cruxcode** [comment]
In the second image that you have uploaded, the command you wrote is wrong. It should be "yarn run build".

**Reply by niteshmodi** [reply]
![q4](https://user-images.githubusercontent.com/89907820/190716843-8f8e1ccd-2348-4cd4-a1a2-92dd87bffcda.png)
even after using yarn run build I am having problems

I even tried making changes ts.config files but still having problems
![q5](https://user-images.githubusercontent.com/89907820/190717070-6c385c29-21cf-4588-a68a-b2d0859cbb1c.png)

**Reply by cruxcode** [reply]
Did you run `lerna run build` and `lerna bootstrap`?

**Reply by adityaxpique** [reply]
Hello Nitesh,
These are the steps to set it up on windows.
1. Clone the repository on your system.
2. Open the terminal in parent folder.
3. Now run `yarn install`.
4. Next you'll have to run `lerna bootstrap`.
5. After that run `lerna run build`.

PS:- You might have to install `lerna` by running `npm i -g lerna`.

**Reply by adityaxpique** [reply]
I would like to add that if you are looking to contribute, you'll have to fork the repository and then follow the above mentioned steps. You will have to set the parent repository as your "upstream" using the command `git remote add upstream https://github.com/Atri-Labs/atrilabs-engine.git`.

**Reply by sanjayk0508** [reply]
Didn't work out on my side. It's showing some errors:

```
info cli using local version of lerna
lerna notice cli v5.3.0
lerna info Executing command in 34 packages: "yarn run build"
lerna info run Ran npm script 'build' in '@atrilabs/manifest-shims' in 25.1s:
yarn run v1.22.19
$ tsc
Done in 22.97s.
lerna info run Ran npm script 'build' in '@atrilabs/react-shim' in 25.6s:
yarn run v1.22.19
$ tsc
Done in 23.56s.
lerna info run Ran npm script 'build' in '@atrilabs/manifest-registry-reader-node' in 26.3s:
yarn run v1.22.19
$ tsc
Done in 24.34s.
lerna info run Ran npm script 'build' in '@atrilabs/forest' in 26.5s:
yarn run v1.22.19
$ tsc
Done in 24.53s.
lerna info run Ran npm script 'build' in '@atrilabs/app-design-forest' in 11.8s:
yarn run v1.22.19
$ tsc
Done in 11.01s.
lerna info run Ran npm script 'build' in '@atrilabs/core' in 12.5s:
yarn run v1.22.19
$ tsc
Done in 11.78s.
lerna info run Ran npm script 'build' in '@atrilabs/component-icon-manifest-schema' in 10.9s:
yarn run v1.22.19
$ tsc
Done in 10.25s.
lerna info run Ran npm script 'build' in '@atrilabs/scripts' in 17.2s:
yarn run v1.22.19
$ tsc
Done in 16.62s.
lerna ERR! yarn run build exited 1 in '@atrilabs/design-system'
lerna ERR! yarn run build stdout:
yarn run v1.22.19
$ build-atri-layer
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
lerna ERR! yarn run build stderr:
'build-atri-layer' is not recognized as an internal or external command,
operable program or batch file.
error Command failed with exit code 1.
lerna ERR! yarn run build exited 1 in '@atrilabs/design-system'
lerna WARN complete Waiting for 4 child processes to exit. CTRL-C to exit immediately.
```

**Reply by adityaxpique** [reply]
Have you executed `lerna bootstrap` followed by `lerna run build`?

**Reply by sanjayk0508** [reply]
Yes. But it's still not working

**Reply by darshitac11** [reply]
@sanjayk0508 Can you please run `lerna bootstrap` and `lerna run build` again? 

There is a known issue with `lerna bootstrap` whose workaround is that you run it a few times if all packages are not built in the first attempt. 

We are planning to move away from `lerna` in the future versions to avoid this problem.

**Reply by sanjayk0508** [reply]
Ok so, this time I ran accordingly - executed `lerna bootstrap` followed by `lerna run build`, executed this multiple times but it still shows the same error:
```
PS C:\Users\Admin\OneDrive\Desktop\Open source\atrilabs\atrilabs-engine> lerna bootstrap
info cli using local version of lerna
lerna notice cli v5.3.0
lerna info bootstrap root only
yarn install v1.22.19
[1/4] Resolving packages...
success Already up-to-date.
Done in 4.52s.
PS C:\Users\Admin\OneDrive\Desktop\Open source\atrilabs\atrilabs-engine> lerna run build
info cli using local version of lerna
lerna notice cli v5.3.0
lerna info Executing command in 34 packages: "yarn run build"
lerna info run Ran npm script 'build' in '@atrilabs/manifest-shims' in 26.6s:
yarn run v1.22.19
$ tsc
Done in 25.55s.
lerna info run Ran npm script 'build' in '@atrilabs/react-shim' in 26.9s:
yarn run v1.22.19
$ tsc
Done in 25.97s.
lerna info run Ran npm script 'build' in '@atrilabs/manifest-registry-reader-node' in 27.3s:
yarn run v1.22.19
$ tsc
Done in 26.50s.
lerna info run Ran npm script 'build' in '@atrilabs/forest' in 28.2s:
yarn run v1.22.19
$ tsc
Done in 27.45s.
lerna info run Ran npm script 'build' in '@atrilabs/app-design-forest' in 12.6s:
yarn run v1.22.19
$ tsc
Done in 11.89s.
lerna info run Ran npm script 'build' in '@atrilabs/core' in 13.9s:
yarn run v1.22.19
$ tsc
Done in 13.26s.
lerna info run Ran npm script 'build' in '@atrilabs/component-icon-manifest-schema' in 15.3s:
yarn run v1.22.19
$ tsc
Done in 14.41s.
lerna info run Ran npm script 'build' in '@atrilabs/scripts' in 23.7s:
yarn run v1.22.19
$ tsc
Done in 22.82s.
lerna ERR! yarn run build exited 1 in '@atrilabs/design-system'
lerna ERR! yarn run build stdout:
yarn run v1.22.19
$ build-atri-layer
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
lerna ERR! yarn run build stderr:
warning You don't appear to have an internet connection. Try the --offline flag to use the cache for registry queries.
'build-atri-layer' is not recognized as an internal or external command,
operable program or batch file.
error Command failed with exit code 1.
lerna ERR! yarn run build exited 1 in '@atrilabs/design-system'
lerna WARN complete Waiting for 4 child processes to exit. CTRL-C to exit immediately.
PS C:\Users\Admin\OneDrive\Desktop\Open source\atrilabs\atrilabs-engine> 
```

**Comment by darshitac11** [comment]
@adityaxpique - please guide @niteshmodi here since you are using Windows.

**Comment by sanjayk0508** [comment]
Even I'm getting errors while setting up, I guess this setup is not suitable for Windows.
OR I'm guessing @niteshmodi didn't activate the windows. I'm saying this because even I didn't activate my windows as well and we both are facing similar issues.
Anyways, It would be great if someone could guide us on this issue.

**Reply by niteshmodi** [reply]
windows activated or not does not matter bro

**Comment by sahilsuman933** [comment, marked as answer]
Hey @niteshmodi, you are facing those issues because the windows shell has different commands. For example: In PowerShell `set` command is used to set paths rather than the `export` command. That's why you are facing these issues. So, I would suggest you install Ubuntu through WSL. 

You can refer to this tutorial https://www.youtube.com/watch?v=2TKh3aokgec. Like it will tell you how to setup ubuntu and how you can integrate it with VS code in your Windows OS.

**Reply by darshitac11** [reply]
This sounds plausible! Thanks for the suggestion @sahilsuman933.

**Comment by darshitac11** [comment]
@all-contributors please add @sahilsuman933 for question.

**Comment by darshitac11** [comment]
@all-contributors please add @sanjayk0508  for bug report

**Comment by devsachin007** [comment]
my atri engine is not starting it is stucking on 
Please wait while we get the server ready...
what to do?

**Reply by darshitac11** [reply]
I have turned your question into a new post and added my reply - https://github.com/Atri-Labs/atrilabs-engine/discussions/331
